src/main/java/edu/ucsb/cs156/gauchoride/controllers/RideController.java

```diff
@@ -39,26 +39,36 @@ public class RideController extends ApiController {
     @Autowired
     UserRepository userRepository;
 
-    @ApiOperation(value = "List all rides")
-    @PreAuthorize("hasRole('ROLE_USER')")
+    @ApiOperation(value = "List all ride requests")
+    @PreAuthorize("hasAnyRole('ROLE_ADMIN','ROLE_DRIVER')")
     @GetMapping("/all")
-    public Iterable<Ride> allRidess() {
+    public Iterable<Ride> allRides() {
         Iterable<Ride> rides = rideRepository.findAll();
         return rides;
     }
 
-    @ApiOperation(value = "Get a ride")
-    @PreAuthorize("hasRole('ROLE_USER')")
+    @ApiOperation(value = "List all of user's ride requests")
+    @PreAuthorize("hasAnyRole('ROLE_USER','ROLE_RIDER')")
+    @GetMapping("/rider")
+    public Iterable<Ride> allRiderRides() {
+        User currentUser =   getCurrentUser().getUser();
+        Iterable<Ride> rides = rideRepository.findAllByRider(currentUser);
+        return rides;
+    }
+
+    @ApiOperation(value = "Get a ride request by its id")
+    @PreAuthorize("hasRole('ROLE_ADMIN')")
     @GetMapping("")
     public Ride getById(
-            @ApiParam("id") @RequestParam Long id) {
+        @ApiParam(name="Ride Request's id", type = "Long", value = "id is a number", example = "10",required = true ) 

```

Comment:
this is unnecessary, when you do <name="Ride Request's id">, it somehow messes up your URL and it looks like this http://localhost:8080/api/rides?Ride%20Request's%20id=1, and it causes 400 response. just do @ApiParam("id") is fine here.

Reply:
Addressed